### ISSUE TITLE
Old PR build remains unexpectedly

### ISSUE THREAD

**Issue by catto** [screwdriver-cd/screwdriver#798]
**What happened**:
Given a pipeline that uses old workflow syntax.
When I create a new PR to the pipeline and migrate a workflow to new one and add `requires: [~pr]` to a job other than `main`,
then `PR-xx:main` PR job remains while this PR is open even if there is a new job named `PR-xx:new-job-name`.

If I reopen this PR, these jobs will be restored and both will run.

**What you expected to happen**:
A PR build with an old name such as `PR-1:main` would be removed or renamed to new one when the workflow was changed.

**How to reproduce it**:

* Create a pipeline like this,
```yaml
jobs:
  main:
    template: foo
```
* open PR without workflow changes
* and change the workflow like this in PR
```yaml
jobs:
  PR:
    requires: [~pr]
    template: foo-pr
  main:
    requires: [~commit]
    template: foo
```
the pipeline page shows `PR-xx:main` and `PR-xx:PR` jobs.
And reopening PR runs both job.

**Comment by d2lam** [comment]
Closing and will track in https://github.com/screwdriver-cd/screwdriver/issues/824
